### PR TITLE
HIP-584: Increase contract create size limit

### LINKED ISSUE #6383: HIP-584: eth_estimateGas fails for contract creates over 6KB
### Description

Currently if we try to perform `eth_estimateGas` for a contract create that uses bytecode exceeding 6KB, we get a validation error and Bad Request. The reason is the web3 module inside Mirror Archive Node has a cap on 6KB for data size.

### Steps to reproduce

1. contract deploy from the hardhat project
2. ethers.js calls eth_estimateGas
3. eth_estimateGas makes a post request to the mirrornode at endpoint contracts/call, with the from, data, and estimate set to true

### Additional context

_No response_

### Hedera network

testnet

### Version

v0.83.0

### Operating system

None

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:

Currently we cap eth_estimateGas requests to have a limit of 6KB for the data size. This causes issues, when we try to estimate contract creates with init bytecode, which exceeds this limit.

The PR fixes this issue and increases the limit to 24KB.

**Related issue(s)**:

Fixes #6383 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java [hunk 1]
```diff
@@ -76,4 +76,16 @@ private boolean hasTo() {
...
+
+    @AssertTrue(message = "must not exceed call size limit")
+    private boolean hasData() {
+        if (data != null) {
+            final var dataSize = data.length();
+
+            // In case of contract calls we should limit requests to 6KiB
+            return to == null || 6144 * 2 > dataSize;
+        } else {
+            return true;
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java:88]: Prefer fail-fast logic in mirror node to reduce nested conditionals e.g. `if (data == null) return true` first.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java [hunk 2]
```diff
@@ -76,4 +76,16 @@ private boolean hasTo() {
...
         return true;
     }
+
+    @AssertTrue(message = "must not exceed call size limit")
+    private boolean hasData() {
+        if (data != null) {
+            final var dataSize = data.length();
+
+            // In case of contract calls we should limit requests to 6KiB
+            return to == null || 6144 * 2 > dataSize;
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java:86]:
```suggestion
            return to == null || dataSize <= 6144 * 2;
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/controller/ContractControllerTest.java [hunk 3]
```diff
@@ -55,6 +55,8 @@
 class ContractControllerTest {
 
     private static final String CALL_URI = "/api/v1/contracts/call";
+    private static final String CONTRACT_BYTES =
```
[the remark this answers is not in the document]
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/controller/ContractControllerTest.java]: Fixed

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/controller/ContractControllerTest.java [hunk 4]
```diff
@@ -224,6 +226,44 @@ void callInvalidValue() {
                 .isEqualTo(new GenericErrorResponse("value field must be greater than or equal to 0"));
     }
 
+    @Test
+    void exceedingDataCallSizOnEstimate() {
```
[the remark this answers is not in the document]
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/controller/ContractControllerTest.java]: Fixed

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java (hunk 1 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java]: Fixed

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java (hunk 2 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java]: Fixed

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/ContractCallRequest.java]: I meant without the unnecessary else statement to reduce nesting, but not a big deal

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**